### ISSUE TITLE
V4: Remap of undefined `columnWrapperStyle` in `FlatList` violates invariant

### ISSUE THREAD

**Issue by cedarbaum** [marklawlor/nativewind#650]
**Describe the bug**
A single column `FlatList` cannot set the `columnWrapperStyle` (see: https://github.com/necolas/react-native-web/blob/master/packages/react-native-web/src/vendor/react-native/FlatList/index.js#L489). When using NativeWind, the style remapper will turn this property into an empty array even when it is undefined, causing the component to crash during construction.

**To Reproduce**
Create any single column `FlatList` component:

```tsx
      <FlatList
        numColumns={1}
        data={DATA}
        renderItem={({ item }) => <Item title={item.title} />}
        keyExtractor={(item) => item.id}
      />

```

**Expected behavior**

CSS remapper preserves undefined for this property, avoiding the invariant violation.

**Comment by cedarbaum** [comment]
As a workaround, `remapProps(FlatList, {...})` can be used and this property can just be removed from remapping. However, I think the default behavior for this property should be to preserve undefined.

**Comment by GustavoBonfimS** [comment]
This problem also happened to me and that's why I couldn't test v4. I'll consider using the workaround that @cedarbaum suggested

**Comment by GustavoBonfimS** [comment]
I created this patch using patch-package to temporarily fix this error. I don't use columnWrapperStyle anywhere in my application so this worked for me.

- patches/react-native-css-interop+0.0.11.patch

```
diff --git a/node_modules/react-native-css-interop/dist/core-components.js b/node_modules/react-native-css-interop/dist/core-components.js
index 1144718..69ffb70 100644
--- a/node_modules/react-native-css-interop/dist/core-components.js
+++ b/node_modules/react-native-css-interop/dist/core-components.js
@@ -56,7 +56,7 @@ exports.unstable_styled = unstable_styled;
     className: "style",
     ListFooterComponentClassName: "ListFooterComponentStyle",
     ListHeaderComponentClassName: "ListHeaderComponentStyle",
-    columnWrapperClassName: "columnWrapperStyle",
+    // columnWrapperClassName: "columnWrapperStyle",
     contentContainerClassName: "contentContainerStyle",
     indicatorClassName: "indicatorStyle",
 });
```

**Comment by feledori** [comment]
Experiencing same issue, thanks for the workaround @GustavoBonfimS.

**Comment by xinha-sh** [comment]
> I created this patch using patch-package to temporarily fix this error. I don't use columnWrapperStyle anywhere in my application so this worked for me.
> 
> * patches/react-native-css-interop+0.0.11.patch
> 
> ```
> diff --git a/node_modules/react-native-css-interop/dist/core-components.js b/node_modules/react-native-css-interop/dist/core-components.js
> index 1144718..69ffb70 100644
> --- a/node_modules/react-native-css-interop/dist/core-components.js
> +++ b/node_modules/react-native-css-interop/dist/core-components.js
> @@ -56,7 +56,7 @@ exports.unstable_styled = unstable_styled;
>      className: "style",
>      ListFooterComponentClassName: "ListFooterComponentStyle",
>      ListHeaderComponentClassName: "ListHeaderComponentStyle",
> -    columnWrapperClassName: "columnWrapperStyle",
> +    // columnWrapperClassName: "columnWrapperStyle",
>      contentContainerClassName: "contentContainerStyle",
>      indicatorClassName: "indicatorStyle",
>  });
> ```

The following worked for me without patch (not sure about perf implications)
```tsx
```ts
export const StyledFlatlist = (props: ComponentProps<typeof FlatList>) => {
  if (props.numColumns === 1) {
    remapProps(FlatList, {
      className: "style",
      ListFooterComponentClassName: "ListFooterComponentStyle",
      ListHeaderComponentClassName: "ListHeaderComponentStyle",
      contentContainerClassName: "contentContainerStyle",
      indicatorClassName: "indicatorStyle",
    });
  } else {
    remapProps(FlatList, {
      className: "style",
      columnWrapperClassName: "columnWrapperStyle",
      ListFooterComponentClassName: "ListFooterComponentStyle",
      ListHeaderComponentClassName: "ListHeaderComponentStyle",
      contentContainerClassName: "contentContainerStyle",
      indicatorClassName: "indicatorStyle",
    });
  }
  return <FlatList {...props} />;
};
```

**Comment by ammarfaris** [comment]
> I created this patch using patch-package to temporarily fix this error. I don't use columnWrapperStyle anywhere in my application so this worked for me.
> 
> * patches/react-native-css-interop+0.0.11.patch
> 
> ```
> diff --git a/node_modules/react-native-css-interop/dist/core-components.js b/node_modules/react-native-css-interop/dist/core-components.js
> index 1144718..69ffb70 100644
> --- a/node_modules/react-native-css-interop/dist/core-components.js
> +++ b/node_modules/react-native-css-interop/dist/core-components.js
> @@ -56,7 +56,7 @@ exports.unstable_styled = unstable_styled;
>      className: "style",
>      ListFooterComponentClassName: "ListFooterComponentStyle",
>      ListHeaderComponentClassName: "ListHeaderComponentStyle",
> -    columnWrapperClassName: "columnWrapperStyle",
> +    // columnWrapperClassName: "columnWrapperStyle",
>      contentContainerClassName: "contentContainerStyle",
>      indicatorClassName: "indicatorStyle",
>  });
> ```

for version 0.0.16 =>
* patches/react-native-css-interop+0.0.16.patch

the file path changes a bit:

```
diff --git a/node_modules/react-native-css-interop/dist/runtime/components/index.js b/node_modules/react-native-css-interop/dist/runtime/components/index.js
index 04f8209..d6bfaf4 100644
--- a/node_modules/react-native-css-interop/dist/runtime/components/index.js
+++ b/node_modules/react-native-css-interop/dist/runtime/components/index.js
@@ -40,7 +40,7 @@ const rendering_1 = require("./rendering");
     className: "style",
     ListFooterComponentClassName: "ListFooterComponentStyle",
     ListHeaderComponentClassName: "ListHeaderComponentStyle",
-    columnWrapperClassName: "columnWrapperStyle",
+    // columnWrapperClassName: "columnWrapperStyle",
     contentContainerClassName: "contentContainerStyle",
     indicatorClassName: "indicatorStyle",
 });

```

**Comment by marklawlor** [comment]
Fixed in `4.0.29`

**Comment by hukpo** [comment]
@marklawlor Can you also add a check for empty string?

Use case is when there is a custom wrapper around FlatList with tailwind-merge. Not critical, but can improve DX a bit.

<img width="649" alt="image" src="https://github.com/marklawlor/nativewind/assets/40897161/3523f1dd-3922-44c8-b4fc-255194b6a904">
